app/lib/utils/gpDataMapper.js

```diff
@@ -3,4 +3,57 @@ function getBookOnlineLink(gpData) {
     ? gpData.onlineServices.appointments.url : undefined;
 }
 
-module.exports = getBookOnlineLink;
+function getFilteredGps(gpData, searchStr) {
+  if (gpData.doctors) {
+    let filteredDocs = [];
+    const searchArr = searchStr.split(' ');
+    searchArr.forEach((searchSubStr) => {
+      const regexp = new RegExp(searchSubStr, 'i');
```

Comment:
I'm using the regex further down to filter the doctors by, however you gave me an idea on how to tidy the `dr` matching up.